application-urlshortener-ui/src/main/resources/URLShortener/WebHome.xml

```diff
@@ -36,5 +36,308 @@
   <minorEdit>false</minorEdit>
   <syntaxId>xwiki/2.1</syntaxId>
   <hidden>true</hidden>
-  <content/>
+  <content>{{warning}}
+{{translation key="urlshortener.duplicates.warning.slow"/}}
+{{/warning}}
+
+----
+
+= {{translation key="urlshortener.duplicates.title"/}} =
+
+{{velocity output="false"}}
+#template('hierarchy_macros.vm')
+## Convenience method.
+#macro (escape $str)
+  $escapetool.xml($services.rendering.escape($str, 'xwiki/2.1'))
+#end
+{{/velocity}}
+
+{{velocity wiki="true"}}
+#set ($query = $services.query.createQuery('property.URLShortener.Code.URLShortenerClass.pageID:*',
+    'solr').setLimit(2000).execute().get(0).getResults())
+## Group shortURL pages by pageID.
+#set ($map = {})
+#foreach ($resultDoc in $query)
+  #set ($shortURLs = $resultDoc.get('property.URLShortener.Code.URLShortenerClass.pageID_string'))
+  #set ($ref = $services.solr.resolveDocument($resultDoc))
+  #set ($ref = $ref.withoutLocale())
+  #foreach ($shortURL in $shortURLs)
+    #if (!$map.keySet().contains($shortURL))
+      #set ($discard = $map.put($shortURL, [$ref]))
+    #elseif (!$map.get($shortURL).contains($ref))
+      #set ($discard = $map.get($shortURL).add($ref))
+    #end
+  #end
+#end
+
+{{html wiki="false"}}
+#set ($noDuplicates = true)
+#foreach ($shortUrl in $map.keySet())
+  #if ($map.get($shortUrl).size() &gt; 1)
+    #set ($noDuplicates = false)
+    &lt;h2&gt;
+      $services.localization.render(
+        'urlshortener.duplicates.title.section',
+        [$escapetool.xml($shortUrl), $map.get($shortUrl).size()]
+      )
+    &lt;/h2&gt;
+    &lt;table&gt;
+    #foreach ($foundDoc in $map.get($shortUrl))
+      &lt;tr&gt;
+        &lt;td&gt;
+          &lt;button class='btn urlshortener-regenerate-btn'
+              data-old-page-id='$escapetool.xml($shortUrl)'
+              data-page-ref='$escapetool.xml($foundDoc)'&gt;
+            $services.localization.render('urlshortener.duplicates.regenerate.button')
+          &lt;/button&gt;
+        &lt;/td&gt;
+        &lt;td style='width: 100%;'&gt;
+          #hierarchy($foundDoc, {})
+        &lt;/td&gt;
+      &lt;/tr&gt;
+    #end
+    &lt;/table&gt;
+  #end
+#end
+{{/html}}
+{{/velocity}}
+
+{{velocity wiki="true"}}
```

Comment:
The above `velocity` macro has the same definition, why do you create a new velocity macro?